contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -3228,104 +3198,43 @@ CREATE OR REPLACE FUNCTION sys.substring(string TEXT, i INTEGER, j INTEGER)
 RETURNS sys.VARCHAR
 AS 'babelfishpg_tsql', 'tsql_varchar_substr' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
 
-CREATE OR REPLACE FUNCTION sys.substring(string NTEXT, i INTEGER, j INTEGER)
-RETURNS sys.NVARCHAR
-AS 'babelfishpg_tsql', 'tsql_varchar_substr' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+CREATE OR REPLACE FUNCTION sys.substring(string PG_CATALOG.BIT, i INTEGER, j INTEGER)
+RETURNS PG_CATALOG.BIT
+AS 'bitsubstr' LANGUAGE INTERNAL IMMUTABLE STRICT PARALLEL SAFE;
 
-CREATE OR REPLACE FUNCTION sys.substring(string sys.VARCHAR, i INTEGER, j INTEGER)
-RETURNS sys.VARCHAR
-AS 'babelfishpg_tsql', 'tsql_varchar_substr' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+CREATE OR REPLACE FUNCTION sys.substring(string BYTEA, i INTEGER, j INTEGER)
+RETURNS BYTEA
+AS 'bytea_substr' LANGUAGE INTERNAL IMMUTABLE STRICT PARALLEL SAFE;
```

Comment:
Why do we need dedicated definitions for BIT/BYTEA, shouldn't we let PG handle them?

Reply:
Removed